hedera-mirror-rosetta/app/persistence/block/block.go

```diff
@@ -159,8 +182,8 @@ func (br *BlockRepository) constructBlockResponse(rf *recordFile, blockIndex int
 	parentHash := rf.PrevHash
 
 	// Handle the edge case for querying first block
-	if rf.PrevHash == genesisPreviousHash {
-		parentIndex = 0          //Parent index should be 0, same as current block index
+	if rf.PrevHash == genesisPreviousHash || parentIndex < 0 {
```

Comment:
We do not need `genesisPreviousHash` anymore. The check for `parentIndex < 0` handles all use-cases if we are populating the `blockIndex` correctly